hangman.rb

```diff
@@ -0,0 +1,255 @@
+require 'artii'
+require 'colorize'
+
+class Hangman
+  attr_accessor :guesses, :word, :hidden_word, :wrongs
+
+  def initialize
+    @guesses = []
+    @word = "" #pick word random from dictionary
+    @hidden_word = []
+    @wrongs = 0
+    @fails = 0
+  end
+
+  def make_hidden_word
+    @word.length.times do @hidden_word << "____  "
+    end
+
+  end
+
+  def choose_word(dictionary)
+    @word = dictionary.sample.upcase
+  end
+
+  def replace_hidden_with_letter(letter)
+    word_index =(0 ... @word.length).find_all {|i| @word[i,1] == letter}
+    num_letters = word_index.length
+
+    num_letters.times do |this|
+      @hidden_word[word_index[this]] = letter
+    end
+  end
+
+
+  def letter_included(guess)
+    if @word.include?(guess)
+      puts "Good Guess"
+      # add letter to hidden_word
+      replace_hidden_with_letter(guess)
+      #################
+      @hidden_word
+    else # ascii minus something
+      @fails += 1
+    end
+  end # End letter_included
+
+
+  def pick_a_letter
+
+    while @word.split("") != @hidden_word
```

Comment:
I like that `pick_a_letter` is its own method!

However, this method ends up being pretty big and complex, even if you don't count the ASCII art. Would it be possible to break it up further? For example, you could have a `get_user_guess` method that asks the user for a letter and verifies it's valid.